meterfs/meterfs.c

```diff
@@ -1203,7 +1206,7 @@ int meterfs_readFile(id_t id, off_t off, char *buff, size_t bufflen, meterfs_ctx
 
 	meterfs_powerCtrl(0, ctx);
 
-	return i;
+	return (err < 0) ? err : i;
```

Comment:
maaaybe we should return err only if `i == 0`? (as we've written `i` bytes to the user-provided buffer?)